libraries/AP_CANManager/AP_CANSensor.cpp

```diff
@@ -142,6 +142,7 @@ bool CANSensor::write_frame(AP_HAL::CANFrame &out_frame, const uint64_t timeout_
 
 void CANSensor::loop()
 {
+    hal.scheduler->delay(10);
```

Comment:
we need to find why this is needed

Reply:
yes, I agree that we need to find why this is required. Although, this delay is only at the startup of the thread, so don't think it affects current operation.

Reply:
I have done a little more testing, it works fine with a 1ms delay too. If I comment the `handle_frame(frame);` it runs with no delay. If I change the new `handle_frame` function to be empty it does not run. So it is might be something to do with the virtual function. Its still not very clear. If I remove `int16_t res = _can_iface->receive(frame, time, flags);` and put `int16_t res = 1` it doesn't run.